From b33fc1d730c2c8b33515c035229f3336f8677ff9 Mon Sep 17 00:00:00 2001
From: Tian Yang <quic_tiany@quicinc.com>
Date: Thu, 27 Jul 2023 00:42:04 -0700
Subject: [PATCH] [net_device] Initialized shadow address for address check

Shadow address needs to be initialized once dev_addr is set.

Change-Id: Ib60de68d148506db7d0fb1f26ee7efe3a1c3a729
Signed-off-by: Tian Yang <quic_tiany@quicinc.com>
---
 net/core/dev.c            | 1 -
 net/core/dev_addr_lists.c | 1 +
 2 files changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/dev.c b/net/core/dev.c
index d9bf57b21080..3a8da4d7fe58 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -1399,7 +1399,6 @@ static int __dev_open(struct net_device *dev, struct netlink_ext_ack *extack)
 	int ret;
 
 	ASSERT_RTNL();
-	dev_addr_check(dev);
 
 	if (!netif_device_present(dev)) {
 		/* may be detached because parent is runtime-suspended */
diff --git a/net/core/dev_addr_lists.c b/net/core/dev_addr_lists.c
index baa63dee2829..3711fdd710e0 100644
--- a/net/core/dev_addr_lists.c
+++ b/net/core/dev_addr_lists.c
@@ -565,6 +565,7 @@ int dev_addr_init(struct net_device *dev)
 		ha = list_first_entry(&dev->dev_addrs.list,
 				      struct netdev_hw_addr, list);
 		dev->dev_addr = ha->addr;
+		memcpy(dev->dev_addr_shadow, dev->dev_addr, MAX_ADDR_LEN);
 	}
 	return err;
 }
-- 
2.34.1

